cmd/podman/create.go

```diff
@@ -764,3 +776,37 @@ func parseCreateOpts(ctx context.Context, c *cli.Context, runtime *libpod.Runtim
 	}
 	return config, nil
 }
+
+func configureRootless(createConfig *cc.CreateConfig, runtime *libpod.Runtime) (bool, int, error) {
```

Comment:
Maybe rename this to make it more obvious what it's going on? `joinPodUserNS` or similar?

Also, do we need something similar to this with normal `podman run` if, for example, `--net=container:$ID` is specified?

Reply:
renamed.

Yes, we would also need to do it in that case (and not allow to join two different containers as that would fail when belonging to different user namespaces).  I'll work on this as a follow up once I am back from PTO.  I've pushed a new version with `configureRootless` -> `joinOrCreateRootlessUserNamespace`